composer/models/resnet/model.py

```diff
@@ -76,9 +82,22 @@ def composer_resnet(model_name: str,
     if initializers is None:
         initializers = []
 
+    # Configure pretrained/weights based on torchvision version
+    if pretrained:
+        weights = 'IMAGENET1K_V2'
+        warnings.warn(
+            'The ``pretrained`` argument for composer_resnet is deprecated and will be removed in the future. Please use ``weights`` instead.'
+        )
+    if weights:
+        pretrained = True
+
     # Instantiate model
     model_fn = getattr(resnet, model_name)
-    model = model_fn(pretrained=pretrained, num_classes=num_classes, groups=groups, width_per_group=width_per_group)
+    model = None
+    if version.parse(torchvision.__version__) < version.parse('0.13.0'):
```

Comment:
Curious if we can avoid parsing the version? Would prefer to try the newer syntax only if it is used. Thinking something along the lines of:
````python

if weights is None:
    if pretrained:
       warnings.warn(DeprecationWarning(...))
    model = model_fn(pretrained=pretrained, num_classes=num_classes, groups=groups, width_per_group=width_per_group)
else:
    if pretrained:
        raise ValueError("Cannot specify pretrained and weights")
    model = model_fn(weights=weights, num_classes=num_classes, groups=groups, width_per_group=width_per_group)
```

Reply:
Using the deprecated path issues a warning, not a TypeError, so unfortunately we can't do that :(